evaluate.sql

```diff
@@ -0,0 +1,247 @@
+--!/usr/bin/env python3
+-- Copyright 2024 Timescale Inc.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+-- http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Repository: https://github.com/timescale/migration-eval
+--
+-- This file runs SQL queries against your database and collects necessary data
+-- to generate a migration evaluation report, which allows us to recommend the
+-- most suitable migration strategy for your specific use case.
+--
+-- To run: psql -t -q -d "<POSTGRES_URI>" -f evaluate.sql
+--
+-- The generated report will be printed in the terminal which can be redirected to a file.
+
+\pset tuples_only
+\set ON_ERROR_STOP on
+
+select version() version \gset
+\echo 'PostgreSQL version:' :version
+
+select pg_size_pretty(pg_database_size(current_database())) db_size \gset
```

Comment:
I think the database size gives us the wrong picture because it includes indexes and materialized views whos data won't be copied during the migration. Should we just include sum of all table_size?

Reply:
Yes, we can do that. However, I think queries that calculate the size of tables individually are expensive to execute and may significantly affect the source database. This might be the reason to avoid such queries. We used to have a similar problem when writing Promscale views that show the size of individual metric tables.

Reply:
@Harkishen-Singh These are approximate numbers and based on size of the table data file on disk. AFAIK, current_database() should do the same behind the scene.